src/wolfio.c

```diff
@@ -473,11 +473,28 @@ int EmbedReceiveFrom(WOLFSSL *ssl, char *buf, int sz, void *ctx)
     recvd = TranslateReturnCode(recvd, sd);
 
     if (recvd < 0) {
+        int err;
         WOLFSSL_MSG("Embed Receive From error");
-        recvd = TranslateIoError(recvd);
-        if (recvd == WOLFSSL_CBIO_ERR_WANT_READ &&
+
+        err = TranslateIoError(recvd);
+        if ((err == WOLFSSL_CBIO_ERR_WANT_READ ||
+                err == WOLFSSL_CBIO_ERR_ISR) &&
```

Comment:
I'm not sure that TCP sockets set `WOLFSSL_CBIO_ERR_ISR` so the commit's changelog is slightly misleading. Nevertheless, I think it's correct to handle `WOLFSSL_CBIO_ERR_ISR` as the `WOLFSSL_CBIO_ERR_WANT_READ`. The only side-effect I see is that we may increase the timeout faster than we should in case of a signal interrupting the recvfrom.

Reply:
We get a `WOLFSSL_CBIO_ERR_ISR` when we reach the timeout on a STREAM socket. I'm not sure why this is instead of a `WOLFSSL_CBIO_ERR_WANT_READ` but that is what is happening here.

Reply:
I tried to test this with a toy c program but I get an `EAGAIN` (that should be translated to `WOLFSSL_CBIO_ERR_WANT_READ`) instead. Can you investigate further?

Reply:
I could not replicate this behaviour. I'm not sure what I was seeing when I wrote this. I'm guessing the debugger might have been messing with the system call. Reverting back to how it was.